notebooks/search-analysis/search_analysis.ipynb

```diff
@@ -0,0 +1,399 @@
+    "plt.ylabel(\"%\")\n",
+    "\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\"\"\"\n",
+    "If you fancy labelling some queries....\n",
+    "\n",
+    "#Get top n search responses for each of our sampled queries\n",
+    "query_sample = query_df.sample(200)\n",
+    "query_sample['top_response']=query_sample['search_term'].map(lambda x:query_search_api(x,5))\n",
+    "\n",
+    "# Futz around with json fields\n",
+    "query_sample = query_sample.explode('top_response')\n",
+    "query_sample['description'] = sample['top_response'].map(lambda x:x.get('description',np.nan) if not isinstance(\n",
+    "                x, float) else [])\n",
+    "query_sample['id'] = sample['top_response'].map(lambda x:x.get('_id',np.nan) if not isinstance(\n",
+    "                x, float) else [])\n",
+    "query_sample['rank'] = sample['top_response'].map(lambda x:x.get('original_rank',np.nan) if not isinstance(\n",
+    "                x, float) else [])\n",
+    "query_sample= query_sample.reset_index(drop=True)\n",
+    "\n",
+    "\n",
+    "from ast import literal_eval\n",
+    "\n",
+    "relevancy_judgments = annotate(query_sample.index,\n",
+    "                               options=['very relevant','somewhat relevant','not relevant','no good answer'],\n",
+    "                               display_fn=lambda idx : display(sample.loc[idx,['search_term','description','id']]))\n",
+    "\n",
+    "query_sample['relevancy']= relevancy_judgments\n",
+    "\n",
+    "query_sample['relevancy'] =  query_sample['relevancy'].map(lambda x: literal_eval(x) if not isinstance(x, float) else np.nan)\n",
+    "score = {'not relevant':0,'somewhat relevant':1,'very relevant':2,'no good answer':np.nan}\n",
+    "query_sample['score'] = query_sample['relevancy'].map(lambda x: score[x[1]])\n",
+    "\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "relevancy_judgements"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#precision at k (k=5)\n",
+    "no_bestbets = relevancy_judgements[relevancy_judgements['best_bet']==False]\n",
+    "precision = no_bestbets[['query_length','session_id','search_term','score']][no_bestbets['score']>1].groupby(['search_term','session_id','query_length']).count()\n",
+    "precision['score'] = precision['score'] / 5\n",
+    "precision = precision.reset_index()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Avg Precision at 5 \n",
+    "\n",
+    "from scipy.stats import sem, t\n",
+    "from scipy import mean\n",
```

Comment:
These two package/function set calls can be placed at the top of the notebook.

Reply:
done-zo